src/main/java/org/opensearch/neuralsearch/search/query/HybridCollectorManager.java

```diff
@@ -215,6 +219,43 @@ private DocValueFormat[] getSortValueFormats(final SortAndFormats sortAndFormats
         return sortAndFormats == null ? null : sortAndFormats.formats;
     }
 
+    private void reduceCollectorResults(
+        QuerySearchResult result,
+        TopDocsAndMaxScore topDocsAndMaxScore,
+        DocValueFormat[] docValueFormats,
+        TopDocs newTopDocs
+    ) {
+        // this is case of first collector, query result object doesn't have any top docs set, so we can
+        // just set new top docs without merge
+        if (result.hasConsumedTopDocs()) {
```

Comment:
what is this case can you please elaborate? I am confused here.

Reply:
hasConsumedTopDocs() will return `true` if topDocs object is null in this QueryResult. Direct getter is weird - it will throw exception in case topDocs is null [code ref](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/search/query/QuerySearchResult.java#L159)

Reply:
updated comment to make i more clear